File: Medium/53-maximum-subarray/try-7.cpp
/*
 * Problem #53: Maximum Subarray
 * Difficulty: Medium
 * Submission: Try 7
 * status: Accepted
 * Language: cpp
 * Date: 04/02/2025, 23:18:53
 * Link: https://leetcode.com/problems/maximum-subarray/
 */

class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int sum = 0,result = INT_MIN;
        for(auto i : nums){
            sum = sum + i;
            result = max(sum,result);
            if(sum<0){
                sum = 0;
            }
        }
        return result;

    }
};

File: Medium/739-daily-temperatures/try-7.py
/*
 * Problem #739: Daily Temperatures
 * Difficulty: Medium
 * Submission: Try 7
 * status: Accepted
 * Language: python3
 * Date: 22/12/2024, 22:33:54
 * Link: https://leetcode.com/problems/daily-temperatures/
 */

class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        stack = []
        n = len(temperatures)
        re = [0]*n
        
        for i in range(n-1,-1,-1):
            while stack and temperatures[stack[-1]]<=temperatures[i] :
                stack.pop()
            re[i] = stack[-1] - i if stack else 0
            stack.append(i)
        return re
